### PR TITLE
fix issue in e2e 

### PR DESCRIPTION
**Pull request by Madhu-1**
remove code duplication in volume_ops_test
call checkFuseAvailable from utils package to avoid code duplication.

after volume mount even if some of the operations fail on the volume. we should umount the volume
calling umount in defer immediately after the volume mount will ensure to call the umount if some operation
fails.
Partial: #1235 

Signed-off-by: Madhu Rajanna <mrajanna@redhat.com>

### CODE REVIEW

File: e2e/volume_ops_test.go [hunk 1]
```diff
@@ -383,9 +383,7 @@ func testVolumeInfo(t *testing.T) {
 }
 
 func testVolumeStatus(t *testing.T) {
-	if _, err := os.Lstat("/dev/fuse"); os.IsNotExist(err) {
-		t.Skip("skipping mount /dev/fuse unavailable")
-	}
+	checkFuseAvailable(t)
```
Comment (harigowtham) [inline e2e/volume_ops_test.go]: I cant find this function. Am i missing anything?

Comment (rishubhjain) [reply e2e/volume_ops_test.go]: https://github.com/gluster/glusterd2/blob/eb548f7cf915b7528cd0605e0f744b3d796782d3/e2e/utils_test.go#L361

File: e2e/snapshot_ops_test.go [hunk 2]
```diff
@@ -314,6 +314,8 @@ func testRestoredVolumeMount(t *testing.T, tc *testCluster) {
 	err := mountVolume(host, snapTestName, mntPath)
 	r.Nil(err, fmt.Sprintf("mount failed: %s", err))
 
+	defer syscall.Unmount(mntPath, syscall.MNT_FORCE|syscall.MNT_DETACH)
```
Comment (prashanthpai) [inline e2e/snapshot_ops_test.go]: Please don't use `syscall.MNT_DETACH`. Here and elsewhere in this PR. These are all glusterfs mounts and not XFS mounts.

Remove it from `e2e/glustershd_test.go` too

**Comment by Madhu-1** [comment]
Fixes: #1156

**Comment by Madhu-1** [comment]
@prashanthpai addressed review comments please review

**Pull request merged**